android/app/src/main/java/app/organicmaps/editor/OsmLoginFragment.java

```diff
@@ -75,12 +95,63 @@ private void login()
     });
   }
 
+  private void loginWithBrowser()
+  {
+    enableInput(false);
+
+    String[] oauthParams = OsmOAuth.nativeOAuthParams();
+    Uri oauth2url = Uri.parse(oauthParams[0] + "/oauth2/authorize")
+            .buildUpon()
+            .appendQueryParameter("client_id", oauthParams[1])
+            .appendQueryParameter("scope", oauthParams[3])
+            .appendQueryParameter("redirect_uri", oauthParams[4])
+            .appendQueryParameter("response_type", "code")
+            .build();
+    try {
+      Intent myIntent = new Intent(Intent.ACTION_VIEW, oauth2url);
+      startActivity(myIntent);
+    } catch (ActivityNotFoundException e) {
+      //Toast.makeText(this, "No application can handle this request."
+      //        + " Please install a webbrowser",  Toast.LENGTH_LONG).show();
```

Comment:
Showing a toast in hard-coded English here is a good idea. And also logging this as a warning.

Reply:
I replaced this code with `Utils.openUri(...)`. It handles missing browser error.